app/models/mixins/supports_feature_mixin.rb

```diff
@@ -265,15 +168,17 @@ def unsupported
 
     # use this for making a class not support a feature
     def unsupported_reason_add(feature, reason = nil)
-      SupportsFeatureMixin.guard_queryable_feature(feature)
+      guard_queryable_feature(feature)
       feature = feature.to_sym
       unsupported[feature] = SupportsFeatureMixin.reason_or_default(reason)
     end
 
-    def define_supports_feature_methods(feature, is_supported: true, reason: nil, &block)
+    def define_supports_feature_methods(feature, is_supported: true, description: nil, reason: nil, &block)
```

Comment:
If possible, and not for this PR, but I would _LOVE_ to get rid of this method.  It creates 2 magic methods per feature, but I can't stand magic methods, especially when the non-magic form is almost the exact same thing.

So, for example.

```ruby
supports :foo
```

creates these 2 magic methods

```ruby
def supports_foo?; ...; end
def self.supports_foo?; ...; end
```

and the callers do

```ruby
Model.new.supports_foo?
```

Why not instead just have a single method that takes a parameter:

```
def supports?(feature); ...; end
```

and change the caller to

```ruby
Model.new.supports?(:foo)
```

Reply:
:+1: no magic

Reply:
💯 👍 👍   Not only that, but creating magic methods adds bloat to the code for no good technical reason.

Reply:
Investigating further, `SupportsFeatureMixin` implements `supports?` as follows

```ruby
  def supports?(feature)
    public_send("supports_#{feature}?")
  end
```

So, we would have to figure out a new approach that does not create all these generated methods.